mantidproject/mantid
# In-place rebinning in the instrument view

**Description of work.**
This PR adds a panel inside the pick tab of the instrument view to allow the user to rebin the workspace without leaving the instrument viewer. Since it replaces the workspace, a couple of bugs have been solved at the same time.

Now, the pick tab also shows the integration over the entire detector if in shape drawing mode and no shape is currently drawn, instead of just showing nothing.

This PR relies on changes from https://github.com/mantidproject/mantid/pull/31754 , do not merge it before this one has been merged first.



**To test:**
- Load a file - preferably an event file like `ExternalData/Testing/Data/SystemTest/CNCS_7860_event.nxs` . 
- Open the instrument viewer, and in the pick tab, change the mode to shape drawing, shape editing or zoom, and check the integration appears and looks correct.
- In the rebin panel under the integration plot, try a few rebin values, and check the plot is correctly updated.
- Finally, checks that saving as histogram does produce a workspace without events.


WIP for #31747 






---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there is GUI work does it follow the [GUI standards](http://developer.mantidproject.org/Standards/GUIStandards.html)?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
